app/javascript/components/mentoring/DiscussionPostForm.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useRef, useState } from 'react'
+import { usePanel } from '../../hooks/use-panel'
+import { MarkdownEditor, MarkdownEditorHandle } from '../common/MarkdownEditor'
+import { sendPostRequest, APIError } from '../../utils/send-request'
+import { useIsMounted } from 'use-is-mounted'
+
+export const DiscussionPostForm = ({
+  endpoint,
+  contextId,
+}: {
+  endpoint: string
+  contextId: string
+}): JSX.Element | null => {
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState<APIError | null>(null)
+  const editorRef = useRef<MarkdownEditorHandle | null>(null)
+  const isMountedRef = useIsMounted()
+  const {
+    open,
+    setOpen,
+    buttonRef,
+    panelRef,
+    componentRef,
+    styles,
+    attributes,
+  } = usePanel()
+
+  const handleSubmit = useCallback(
+    (e) => {
+      e.preventDefault()
+
+      setIsLoading(true)
+
+      sendPostRequest({
+        endpoint: endpoint,
+        body: { content: editorRef.current?.getValue() },
+        isMountedRef: isMountedRef,
+      })
+        .then(() => setOpen(false))
+        .catch((err) => {
+          if (err instanceof Response) {
+            err.json().then((res: any) => {
+              setError(res.error)
+            })
+          }
+        })
+        .finally(() => setIsLoading(false))
+    },
+    [editorRef, endpoint, isMountedRef, setOpen]
+  )
+
+  const handleEditorMount = useCallback(
+    (editor) => {
+      editorRef.current = editor
+    },
+    [editorRef]
+  )
+
+  return (
+    <div ref={componentRef}>
+      <button
+        ref={buttonRef}
+        onClick={() => {
+          setOpen(!open)
+        }}
+        type="button"
+      >
+        Add a comment
+      </button>
+      <div ref={panelRef} style={styles.popper} {...attributes.popper}>
+        {open ? (
+          <div>
+            <form onSubmit={handleSubmit}>
+              <MarkdownEditor
+                editorDidMount={handleEditorMount}
+                contextId={contextId}
+              />
+              <button type="submit" disabled={isLoading}>
+                Send
+              </button>
+            </form>
```

Comment:
Something that's not clear from the designs maybe is that this form sits *above* the panel. So whichever tab you're on the form still sits at the bottom of the page (e.g. you can be reading your scratchpad while typing things into the box at the bottom).

Reply:
So you mean that it floats on the bottom of the page? If that's the case, should the panel not close then when the user clicks around the page? If yes, I think we need a close button on the panel, too

Reply:
I'm not sure what you mean by the "panel' and what you mean by "close". 

- I'm guessing you mean the RHS by the panel? 
- I can't come up with what "close" means here though.

My understanding:
- Clicking a different iteration in the header should basically re-render all the things below.
- Once someone clicks on the button at the bottom of the RHS it opens up a text-area that hovers above the the RHS (at the bottom) 

I'm on Slack if you want to discuss it all more :)

Reply:
Marking this as resolved, as we've talked about this now. I'm going to implement the correct placement of the textbox in a separate PR.